### DISCUSSION TITLE
Simplify the initial set up

### DISCUSSION THREAD

**Discussion by jbesclapez** [inovex/scrumlr.io#4452] [Q&A]
Hello,

First a big thank you to the community. It is a really nice project in terms of functionalities. And you guys, managed to make it visually appealing too. Great job!

I have many docker containers and regularly install and test containers with new solutions. See me as a docker consumer with advanced knowledge but far from a real dev. I have recently spent days customizing this repo to my needs.

Here is the main problem of project : It is not simple to start using.
- There is no official docker-hub official container.
- After cloning, there is no default docker-compose in the main folder. Like in many other repos.
- After cloning, there is no default env file. Both those files are not simple to find when you do not know...
- Even after, using the recommended ENV and Docker-compose, it is not working out of the box. (https://github.com/inovex/scrumlr.io/tree/8c3439f162c96ab42a8afc995107ced956c2c659/deployment/docker)
- Changing the ports becomes complex.
- Adding HTTPS becomes complex.
At the end you end up rewriting, some lines or codes, and it makes the emboarding of new comer difficult. This app would be really more successful (compared to its competitors) if we could simplify its emboarding.

Could someone update the repo with a default working ENV and COMPOSE that would work out of the box?

Example : I am using the default docker compose and ENV that I changed but i get an empty page even if the docker started. However, the errors in the browser are : 
1- Cookie Warnings.
2- Loading failed for the <script> with source “http://localhost:3000/static/js/main.3073a907.js”.

I do not map any port 3000, so I need to find where it is used.

Also, dont get me wrong, I managed to make it work on my own repo but It was far from simple. And that is my point. How can we simplify that...? I am sure this project is the best retro tool on the market but its adoption is difficult. 

I would be happy to help and talk with a DEV, so please PM me. :-)

Thanks

**Comment by Lennart01** [comment]
> There is no official docker-hub official container.

DockerHub puts arbitrary restrictions on [pulls](https://docs.docker.com/docker-hub/download-rate-limit/).
In theory the container registry we publish to doesn't matter to the end user.

> Changing the ports becomes complex.

Changing ports is super easy. You don't need to change the container port and only the host port.

> Adding HTTPS becomes complex.

Handling HTTPS is not intended to be done from inside of the container.
We strongly recommend using a reverse proxy in front which handles TLS termination.

> Even after, using the recommended ENV and Docker-compose, it is not working out of the box. 

The docs page is not finalized and I agree that the premade compose sample might need some more tuning (rn we dont use registry images yet bec we didnt support ARM earlier).

**Comment by BlackPenguins** [comment]
Yup. I agree with OP. First-time user, and software engineer here with general knowledge of docker-compose. Dockers are supposed to be super easy to run right out of the box. I have been struggling to get this working and keep getting this error:

`scrumlr-backend_1   | {"level":"info","ts":1726688772.2296185,"caller":"migrations/migrations.go:44","msg":"successfully migrated database"}
scrumlr-backend_1   | {"level":"fatal","ts":1726688772.229906,"caller":"scrumlr.io/main.go:246","msg":"failed to connect to nats message queue: unable to connect to nats server \"nats://nats:4222\": parse \"\\\"nats://nats:4222\\\"\": first path segment in URL cannot contain colon","stacktrace":"main.run\n\t/go/src/github.com/inovex/scrumlr.io/main.go:246\ngithub.com/urfave/cli/v2.(*Command).Run\n\t/go/pkg/mod/github.com/urfave/cli/v2@v2.27.4/command.go:276\ngithub.com/urfave/cli/v2.(*App).RunContext\n\t/go/pkg/mod/github.com/urfave/cli/v2@v2.27.4/app.go:333\ngithub.com/urfave/cli/v2.(*App).Run\n\t/go/pkg/mod/github.com/urfave/cli/v2@v2.27.4/app.go:307\nmain.main\n\t/go/src/github.com/inovex/scrumlr.io/main.go:218\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}
`

So it cannot connect to nats I assume, whatever that is. But I just ran the docker-compose as is so it should be already configured correctly.

That and the front-end is complaining about an 'h' somewhere? Almost like a typo:

`scrumlr-frontend_1  | 2024/09/18 19:49:31 [emerg] 1#1: unexpected "h" in /etc/nginx/conf.d/scrumlr.io.conf:36
`

I will be messing around with this until it works but this is NOT friendly to first-time users.

I also left most of the default .env the same except for the database password. Maybe I'm supposed to be configuring more than that. But the documentation doesn't help much.

**EDIT:** Reverted everything. Upgraded my docker-compose and docker. I am farther now. The page is trying to load on localhost:9090 but now I'm getting 404 errors with various js files.
`scrumlr-frontend-1  | 2024/09/19 16:01:51 [error] 29#29: *1 open() "/usr/share/nginx/html/inovex/scrumlr.io/static/css/main.4ab6465e.css" failed (2: No such file or directory), client: 192.9.200.181, server: _, request: "GET /inovex/scrumlr.io/static/css/main.4ab6465e.css HTTP/1.1", host: "pazuzu.lab.rocsoft.com:9090", referrer: "http://pazuzu.lab.rocsoft.com:9090/"
`

**Reply by Lennart01** [reply]
Hey thanks for your feedback. We will look into this.

**Comment by Lennart01** [comment]
I have tried to address these issues in #4521.
Deployment should now work out of the box.
I have removed the building of local images to use our prebuilt images (we now have ARM images available).
A caddy deployment is also now included out of the box to improve your self hosting experience.
